python/mpt/src/mpt/trie_diff.cairo

```diff
@@ -90,8 +98,96 @@ struct AddressAccountNodeDictAccess {
     new_value: AccountNode,
 }
 
-// Process the difference between two account leaf nodes
-func _process_account_diff{}(path: Bytes32, left: LeafNode, right: LeafNode) -> () {
+// @notice Decode the RLP encoded representation of an account node.
+func AccountNode_from_rlp{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(
+    encoding: Bytes
+) -> AccountNode {
+    alloc_locals;
+
+    let decoded = decode(encoding);
+
+    let sequence = decoded.value.sequence;
+    let len = sequence.value.len;
+    let data = sequence.value.data;
+
+    let nonce_bytes = data[0].value.bytes;
+    let balance_bytes = data[1].value.bytes;
+    let storage_root_bytes = data[2].value.bytes;
+    let codehash_bytes = data[3].value.bytes;
+
+    let balance = U256_from_be_bytes(balance_bytes);
+    let codehash = Bytes_to_Bytes32(codehash_bytes);
+    let nonce = Uint_from_be_bytes(nonce_bytes);
+    let storage_root = Bytes_to_Bytes32(storage_root_bytes);
+
+    tempvar res = AccountNode(
+        new AccountNodeStruct(
+            nonce=nonce, balance=balance, code_hash=codehash, storage_root=storage_root
+        ),
+    );
+
+    return res;
+}
+
+// Process the difference between two account f nodes
+func _process_account_diff{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+    address_preimages: MappingBytes32Address,
+    main_trie_end: AddressAccountNodeDictAccess*,
+}(path: Bytes32, left: OptionalLeafNode, right: OptionalLeafNode) -> () {
+    alloc_locals;
+    let dict_ptr = cast(address_preimages.value.dict_ptr, DictAccess*);
+
+    let fp_and_pc = get_fp_and_pc();
+    local __fp__: felt* = fp_and_pc.fp_val;
+
+    let (pointer) = hashdict_read{dict_ptr=dict_ptr}(2, path.value);
+    let new_dict_ptr = cast(dict_ptr, Bytes32OptionalAddressDictAccess*);
+    tempvar address_preimages = MappingBytes32Address(
+        new MappingBytes32AddressStruct(address_preimages.value.dict_ptr_start, new_dict_ptr)
+    );
+    tempvar address = Address(pointer);
+
+    if (left.value != 0) {
+        let left_decoded = AccountNode_from_rlp(left.value.value.value.bytes);
```

Comment:
i laughed at the .value.value.value.value ;D

Reply:
CURSED